server/src/main/java/io/crate/execution/engine/aggregation/impl/TopKAggregation.java

```diff
@@ -198,6 +216,77 @@ public DataType<?> partialType() {
         return new StateType(boundSignature.argTypes().getFirst());
     }
 
+    @Nullable
+    @Override
+    public DocValueAggregator<?> getDocValueAggregator(LuceneReferenceResolver referenceResolver,
+                                                       List<Reference> aggregationReferences,
+                                                       DocTableInfo table,
+                                                       List<Literal<?>> optionalParams) {
+        if (aggregationReferences.size() != 1) {
+            return null;
+        }
+        Reference reference = aggregationReferences.get(0);
+        if (!reference.hasDocValues()) {
+            return null;
+        }
+
+        int limit;
+        if(optionalParams.size() == 1) {
+            limit = (int) optionalParams.getFirst().value();
+        } else {
+            limit = DEFAULT_LIMIT;
+        }
+        return getDocValueAggregator(reference, limit);
+    }
+
+    @Nullable
+    private DocValueAggregator<?> getDocValueAggregator(Reference ref, int limit) {
+        if (!ref.hasDocValues()) {
+            return null;
+        }
+        return switch (ref.valueType().id()) {
+            case ByteType.ID,
+                 ShortType.ID,
+                 IntegerType.ID,
+                 LongType.ID,
+                 TimestampType.ID_WITH_TZ,
+                 TimestampType.ID_WITHOUT_TZ,
+                 FloatType.ID,
+                 DoubleType.ID,
+                 GeoPointType.ID ->
```

Comment:
I think GeoPoint would need some additional decode logic and won't work as is.
See https://github.com/crate/crate/blob/c8f7d8313d6e2a5aa252f50187a8bfb7f419389a/server/src/main/java/io/crate/expression/reference/doc/lucene/GeoPointColumnReference.java#L53-L58

(This also means we should probably use `DataTypeTesting.randomType()` or iterate over all types in at least one of the test cases, to really execute all types)

Reply:
+1, see `test_random_type_random_values` in `HyperLogLogDistinctAggregationTest`

Reply:
This will also be a follow-up for string/longs it not really needed.